### PR TITLE
[Gateway] : Test Setup Jest / Cypress

### LINKED ISSUE #201: [gateway] Gateway testing
https://github.com/omgnetwork/optimism-v2/issues/35

Add a functional test (with CI - gh actions integration) for the gateway:

Perhaps use Robotframework 
https://medium.com/@joonasvenlinen/ui-regression-testing-with-robot-framework-part-1-578f2f50b721

### PR DESCRIPTION
**Pull request by sk-enya**
:clipboard: Add associated issues, tickets, docs URL here.

## Overview

Added support for jest to bump up the test cases for the Gateway.

## Changes

- Setting up the jest configuration for a gateway to bump up the unit testing for each module and component of the app.
- Also add a setup for cypress which will validate the gateway after the build.
- Added sample snapshot test for `layout/index.js`

## Testing
execute from root `yarn && yarn clean`
to run the test cases execute from gateway `yarn test`
to build snapshots execute `yarn snapshots`

### CODE REVIEW

File: packages/boba/gateway/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
```
Comment (wsdt) [inline packages/boba/gateway/babel.config.js]: why do we have a babel config now? Did we eject the React app?

**Review by wsdt — APPROVED** [review]
left a comment, but looks good to me!

File: packages/boba/gateway/babel.config.js (hunk 1 above)
Comment (sk-enya) [reply packages/boba/gateway/babel.config.js]: Aahh, Yeah we haven't `eject` react app yet... 

Fixed the babel as it's not required.

Good catch thank you!

File: packages/boba/gateway/readme.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
 # Gateway
 gateway.boba.betwork.
 
+## To Run the app in dev mode.
+`yarn start`
+
+## To run the unit tests
+`yarn test`
+
+## To prepare the snapshots
```
Comment (InoMurko) [inline packages/boba/gateway/readme.md:10]: what are snapshots?

Comment (sk-enya) [reply packages/boba/gateway/readme.md:10]: `Snapshots` is way to make sure that there is no unexpected changes in UI `like html markup which is getting generated for component`

**Comment by sk-enya** [comment]
Closing this as need to be implemented in better way as cypress doesn't have direct support to meta mask..

Handling this in separate pr with synpress

**Pull request closed without merging**